eservice/pdo/eservice/pdo_enclave.py

```diff
@@ -233,7 +242,7 @@ def get_enclave_service_info(spid, config=None) :
     logger.debug("Attempting to load enclave at: %s", signed_enclave)
 
     num_of_enclaves = 1
-    pdo = enclave.pdo_enclave_info(signed_enclave, spid, num_of_enclaves)
+    pdo = enclave.pdo_enclave_info(signed_enclave, spid, attestations_type, num_of_enclaves)
```

Comment:
```suggestion
    pdo = enclave.pdo_enclave_info(signed_enclave, spid, attestation_type, num_of_enclaves)
```

Reply:
Good catch!
This error was not detected during tests because this piece of code is not tested in this PR, but it is tested in #380 .
Hence, I'm going to covert this to a draft until the other is merged (and this is rebased).

(Long explanation, for the records)

The registration of the enclave info and IAS keys on the ledger is performed through the `register-with-ledger.sh` script. This is only triggered in Sawtooth builds in HW mode, since CCF builds do not support HW mode currently.
This is visible in the `docker/Makefile`, specifically in the `make [..] register` occurrences.

The removal of Sawtooth implies the removal of the tests of the registration procedure -- which uses the python code for grabbing an enclave attestation to get the mrenclave and basename. This would result in a lower test coverage.

For this reason, #380 adds a `make [..] register` occurrence in the `test-env-setup-with-no-build-ccf` section.
This makes sure that the script is still triggered in HW mode.
This is important in conjunction with #379 , which allows to execute (only) the eservice in HW mode.
The script does not trigger errors since the interaction with the ledger is removed (again, the CCF TP does not support this feature), but provides warnings to remind to fix this once CCF TP support is completed.

In #382 , the script is further refined to trigger the registration only in HW mode when EPID is selected.

Reply:
As expected, with #380 merged, this issue showed up in the tests, and even a few more.
These have been addressed in https://github.com/hyperledger-labs/private-data-objects/pull/382/commits/bff1abc7f5de7b48f6b09e699adf2402b877ea4b